### PR TITLE
Make `SourceState` similar to `MigrateState`

### PR DESCRIPTION
**Pull request by FourTOne5**

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/SourcePresenter.kt [hunk 1]
```diff
@@ -28,23 +28,24 @@ class SourcePresenter(
...
+                .collectLatest { sourceList ->
+                    val uiModels = sourceList.toSourceUiModels()
+                    _state.emit(SourceState.Success(uiModels))
                 }
-                .collectLatest(::collectLatestSources)
         }
     }
 
-    private fun collectLatestSources(sources: List<Source>) {
+    private fun List<Source>.toSourceUiModels(): List<SourceUiModel> {
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/SourcePresenter.kt:48]: Why the change to an extension function?

Comment (FourTOne5) [reply app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/SourcePresenter.kt:48]: just felt like it...

Comment (ghostbear) [reply app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/SourcePresenter.kt:48]: You could make `collectLatestSources` a suspend function rather than making an extension function. As I guess this change is in part with having to use `_state.emit`
